### ISSUE TITLE
Issues with using 1.21

### ISSUE THREAD

**Issue by ShFil119** [kcat/openal-soft#490]
```
In file included from /usr/include/AL/alext.h:29,
                 from ../src/audio/oal/channel.h:6,
                 from ../src/audio/oal/channel.cpp:4:
/usr/include/c++/10.2.0/cinttypes:58:11: error: ‘imaxdiv_t’ has not been declared in ‘::’
   58 |   using ::imaxdiv_t;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cinttypes:61:11: error: ‘imaxabs’ has not been declared in ‘::’
   61 |   using ::imaxabs;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cinttypes:62:11: error: ‘imaxdiv’ has not been declared in ‘::’
   62 |   using ::imaxdiv;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cinttypes:68:11: error: ‘strtoimax’ has not been declared in ‘::’
   68 |   using ::strtoimax;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cinttypes:69:11: error: ‘strtoumax’ has not been declared in ‘::’
   69 |   using ::strtoumax;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cinttypes:72:11: error: ‘wcstoimax’ has not been declared in ‘::’
   72 |   using ::wcstoimax;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cinttypes:73:11: error: ‘wcstoumax’ has not been declared in ‘::’
   73 |   using ::wcstoumax;
```
@kcat, do you think it's on gcc's side?

**Comment by kcat** [comment]
I would appear to be a GCC issue. `alext.h` includes the standard `cinttypes` header in C++11 and later to get the `[u]int64_t` types, and standard headers like that are supposed to be self-sufficient (it should be usable as-is without having to include other things first). It would seem to be a problem in its headers if it's trying to use things that it hasn't made sure are declared.

**Comment by ShFil119** [comment]
Reported:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=97728

**Comment by ShFil119** [comment]
hmm, similar stuff with libc++:
```
/usr/bin/../include/c++/v1/cinttypes:247:8: error: no member named 'imaxdiv_t' in the global namespace
using::imaxdiv_t;
     ~~^
/usr/bin/../include/c++/v1/cinttypes:248:8: error: no member named 'imaxabs' in the global namespace
using::imaxabs;
     ~~^
custompipes_gl.cpp
/usr/bin/../include/c++/v1/cinttypes:249:8: error: no member named 'imaxdiv' in the global namespace
using::imaxdiv;
     ~~^
/usr/bin/../include/c++/v1/cinttypes:250:8: error: no member named 'strtoimax' in the global namespace
using::strtoimax;
     ~~^
/usr/bin/../include/c++/v1/cinttypes:251:8: error: no member named 'strtoumax' in the global namespace
using::strtoumax;
     ~~^
/usr/bin/../include/c++/v1/cinttypes:252:8: error: no member named 'wcstoimax' in the global namespace; did you mean 'wcstombs'?
using::wcstoimax;
``

**Comment by jwakely** [comment]
This suggests you have a header called `inttypes.h` on your system which is being found instead of the standard one.

@kcat:
> `alext.h` includes the standard `cinttypes` header in C++11 and later to get the `[u]int64_t`

You only need `<cstdint>` for those types, not `<cinttypes>`.

**Comment by kcat** [comment]
> You should probably just include `<stdint.h>` for all platforms instead of the [preprocessor dance](https://github.com/kcat/openal-soft/blob/c0cbe602ce66481a822ba16856052a47154f5cc5/include/AL/alext.h#L25) that isn't right.

True. For some reason I was under the impression `stdint.h` wasn't technically valid under C++ and only worked because C++ compilers also tend to have access to C compiler headers. But I double-checked that and it's not true, C++ does specify `stdint.h`'s existence. That does thankfully simplify things. Fixed that in commit 40c92c4643193d09c2fd90a5b3df4fe57230a789.

**Comment by madebr** [comment]
I think this issue can be closed.
The issue was that the project uses a header `inttypes.h` that clashed with the one from the c standard.
https://github.com/GTAmodding/re3/pull/870
